Port forwarding fails and doesn't recover when hitting os max open files limit

### Vagrant version

`2.2.18`

### Vagrant VMware plugin version

`3.0.0`

### Vagrant VMware utility version

`1.0.20`

### Host operating system

MacOS Big Sur 11.5.2

### Guest operating system

NixOS

### Vagrantfile

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "esselius/nixos"
  config.vm.box_version = "21.05"

  config.vm.provider :vmware_desktop do |v|
    v.vmx["memsize"] = 10240
  end

  config.vm.network "forwarded_port", guest: 80, host: 80
  config.vm.network "forwarded_port", guest: 443, host: 443
  config.vm.network "forwarded_port", guest: 6443, host: 6443
end
```

### Expected behavior

Port forwarding keeps working

### Actual behavior

Local port is closed and connections are refused until recreating VM.

### Steps to reproduce

1. Create Vagrant VM with forwarded ports
2. Open enough tcp-connections to hit service maxfiles-limit (default 256)
3. Try to use forwarded port

I'm running a k8s-cluster in the VM and use the helm terraform provider to deploy some charts, but even when using `terraform apply -parallelism=1`, the number of connections easily hits the limit.

```
2021-09-21T17:22:12.527+0200 [ERROR] vagrant-vmware-utility.api.pfwd-service.fwd: failed to accept incoming connection: fwd="&{true context.Background.WithCancel 0xc000322c40 0x118d140 {0 0} 0xc0003610e0}" error="accept tcp [::]:6443: accept: too many open files"
```

Error handling code: https://github.com/hashicorp/vagrant-vmware-desktop/blob/946a313688312589240c8423e5b627bf83dbaae2/go_src/vagrant-vmware-utility/internal/service/port_forwarding.go#L57-L62

While loop checking number of open files during terraform-run:

```
$ while true                                                                                     
    sudo lsof -p 494|wc -l
    sleep 1
  end
19
19
19
19
29        # Running terraform apply
85
103
103
103
103
105
134
134
134
134
134
136
134
134
134
134
134
134
136
130
130
130
130
130
130
130
130
130
130
130
130
130
138
150
184
196
208
208
208
210
208
208
208
236
236
236
236
236
236
236
236
236
236
236
236
236
236
236
246
23         # Port forwarding fails
23
23
23
```

Terraform error

```
Plan: 3 to add, 0 to change, 0 to destroy.
helm_release.ingress-nginx: Creating...
helm_release.ingress-nginx: Still creating... [10s elapsed]
helm_release.ingress-nginx: Still creating... [20s elapsed]
helm_release.ingress-nginx: Still creating... [30s elapsed]
helm_release.ingress-nginx: Still creating... [40s elapsed]
helm_release.ingress-nginx: Creation complete after 45s [id=ingress-nginx]
helm_release.local-path-provisioner: Creating...
helm_release.local-path-provisioner: Still creating... [10s elapsed]
helm_release.local-path-provisioner: Creation complete after 13s [id=local-path-provisioner]
helm_release.clickhouse-operator: Creating...
╷
│ Error: rendered manifests contain a resource that already exists. Unable to continue with install: could not get information about the resource: Get "https://127.0.0.1:6443/api/v1/namespaces/clickhouse-operator/serviceaccounts/clickhouse-operator-altinity-clickhouse-operator": dial tcp 127.0.0.1:6443: connect: connection refused
│
```

I figured I could just use the VM IP directly, but that quickly gets messy when also generating your own k8s tls certs etc :)

Maybe it's just a matter of adding a few lines to the generated launchd service config

--- a/go_src/vagrant-vmware-utility/command/service_darwin.go
+++ b/go_src/vagrant-vmware-utility/command/service_darwin.go
@@ -53,6 +53,15 @@ const LAUNCHD_JOB = `<?xml version="1.0" encoding="UTF-8"?>
         <string>%s</string>
     <key>AbandonProcessGroup</key>
         <true/>
+	  <!--
+	    default is 256 which can be low if we are handling port
+	    forwarding so increase to a reasonably larger amount
+	  -->
+	  <key>SoftResourceLimits</key>
+	  <dict>
+	      <key>NumberOfFiles</key>
+	      <integer>4096</integer>
+	  </dict>
 </dict>
 </plist>
 `

